stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1087,6 +1088,17 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         _sliceRange = range
     }
     
+    @inline(__always)
+    func _validateIndex(_ index: Int, message: String? = nil) {
+        precondition(_sliceRange.contains(index), message ?? "Index \(index) is out of bounds of range \(_sliceRange)")
+    }
+    
+    func _validateRange<R: RangeExpression>(_ range: R) where R.Bound == Int {
```

Comment:
LGTM overall. Just wondering — is the cost of inlining `_validateRange` particularly high, or is it missing `@inline(__always)` for an unrelated reason?